src/onediff/infer_compiler/with_oneflow_compile.py

```diff
@@ -349,6 +349,31 @@ def load_graph(self, file_path, device=None, run_warmup=True):
     @cost_cnt(transform_mgr.debug_mode)
     def save_graph(self, file_path):
         state_dict = self.runtime_state_dict()
+
+        import oneflow.framework.args_tree as args_tree
+
+        def disabled_dataclass(value):
+            return False
+
+        original_is_dataclass = args_tree._is_dataclass
+        args_tree._is_dataclass = disabled_dataclass
+
+        import dataclasses
+
+        def reverse_dataclass(value):
+            if dataclasses.is_dataclass(value):
+                return reverse_proxy_class(type(value))(**value)
+            else:
+                return value
+
+        for name, rsd in state_dict.items():
+            output = state_dict[name]["outputs_original"]
+            out_tree = args_tree.ArgsTree((output, None), False)
+            out = out_tree.map_leaf(reverse_dataclass)
```

Comment:
```suggestion
            # dataclass type needs to be reversed to torch type to avoid saving error.
            out = out_tree.map_leaf(reverse_dataclass)
```